### DISCUSSION TITLE
Github Pages site visit 404

### DISCUSSION THREAD

**Discussion by guowei-gong** [community/community#47342] [Other Feature Feedback, Questions, & Ideas] [answered]
## Problem 
I have deployed the website through github page, custom domain, and the articles are accessible normally. But copy and paste the article address, open a new page in the browser, paste the article address, the page 404. I can confirm through the text comparison tool that they have exactly the same link address, after accessing the home page through the custom domain, click on the article can be accessed, ** directly accessing the article (not accessing the index.html home page of the custom domain), will 404. this 404 page is not This 404 page is not the 404 page of the theme, but the 404 page of the github page**.
### Replication
1. visit [guowei-gong.github.com](https://guowei-gong.github.io/)
2. click on any article
3. copy the address of the article (either directly from the address bar or by copying the article address at the end of the article)
4. Open a new window in your browser
5. Paste the address of the article and reproduce it
### My attempts
I tried to find out if anyone else was experiencing the same thing as me and came across one of the 2015 discussions (Fargo discussion) where they thought it had something to do with the repository of the github page, if that repository is your username then it always resolves the `main` branch and the other repositories resolve the `gh-pages` branch. I'm currently using a repository named after my username before I cry, and I've changed the Branch under Repositories -> Settings -> Pages, as shown below. Could this be the cause of this problem? I'd like to try deploying directly to the `main` branch as a next step.

![image](https://user-images.githubusercontent.com/62499904/218695578-e9349abb-ddff-414c-9110-3679a49d9eae.png)

### Supplementary
![image](https://user-images.githubusercontent.com/62499904/218905819-8e82448f-daa3-4c1e-bd24-0550a9d01658.png)

Now I've tried deploying in the main branch and it's been 15 hours and still the same problem!
I cancelled the Github action workflow deployment and pushed the locally generated static page directly, and I was able to make sure that the static page looks fine locally. [Here](https://github.com/guowei-gong/guowei-gong.github.io) is the github repo address

**Comment by jge162** [comment]
why don't you try to deploy it from the main branch instead?

**Reply by jge162** [reply]
Also after you change to main branch, expect a delay. It takes time to deploy.

**Reply by guowei-gong** [reply]
Thank you for your reply. I tried deploying in the main branch and 15 hours have passed and still the same problem!

![image](https://user-images.githubusercontent.com/62499904/218904979-76de6f50-802b-44c6-9972-1aa503f4a8c1.png)

**Comment by guowei-gong** [comment, marked as answer]
**Currently I use hugo to generate static pages without this problem.** It is possible that this is a problem with the static generation framework?

**Reply by jge162** [reply]
not sure tbh

**Reply by guowei-gong** [reply]
Here is the static page generated by hugo.
![image](https://user-images.githubusercontent.com/62499904/218968457-f6a66758-d356-4b83-ae28-ea640e88f3d6.png)

To access the article, the address bar is where the red arrow points, not the article in the `post` folder generated by the other framework before.

**Comment by aadictive** [comment]
@guowei-gong Where you able to resolve this using the GitHub pages?
